internal/pkg/runtime/launcher/oci/process_linux.go

```diff
@@ -32,7 +32,9 @@ const singularityLibs = "/.singularity.d/libs"
 // Set Singularity> prompt, try bash --norc, fall back to sh.
 var ociShellScript = "export PS1='Singularity> '; test -x /bin/bash && exec /bin/bash --norc || exec /bin/sh"
 
-func (l *Launcher) getProcess(ctx context.Context, imgSpec imgspecv1.Image, bundle string, ep launcher.ExecParams, u specs.User) (*specs.Process, error) {
+// getProcess creates and returns an specs.Process struct defining the execution behavior of the container.
+// The userEnv map returned also holds all user-requested environment variables (i.e. not those from the image).
+func (l *Launcher) getProcess(ctx context.Context, imgSpec imgspecv1.Image, bundle string, ep launcher.ExecParams, u specs.User) (process *specs.Process, userEnv map[string]string, err error) {
```

Comment:
We should at some point (not as part of this PR, probably) do something about this overly-large method signature...